core/keychain/keychain.go

```diff
@@ -366,3 +370,24 @@ func (kc *keychain) retrieveKeyPair() (privKey []byte, mnemonic string, err erro
 
 	return privKey, mnemonic, nil
 }
+
+func (kc *keychain) GetManagedThreadKey(threadKeyName string) (thread.Key, error) {
+	size := 32
+
+	privBytes, _, err := kc.retrieveKeyPair()
```

Comment:
Seeing @dmerrill6 's other PR reminded me we should use `GetStoredKeyPairInLibP2PFormat` here as well so it checks the cache first under the hood.

Reply:
@jsonsivar 

Thank you! That's the reason for the slowness I noticed then.

f0b4486a678384dc96c4ce8af3d70f8ebb3aa927